emote/nn/gaussian_policy.py

```diff
@@ -61,22 +64,25 @@ def forward(self, x: Tensor) -> Tuple[Tensor, float]:
 
         mean = self.mean(x).clamp(min=-5, max=5)  # equates to 0.99991 after tanh.
         std = torch.exp(self.log_std(x).clamp(min=-20, max=2))
+        if self.training:
+            dist = dists.TransformedDistribution(
+                dists.Independent(dists.Normal(mean, std), 1),
+                RobustTanhTransform(),
+            )
+            sample = dist.rsample()
 
-        dist = dists.TransformedDistribution(
-            dists.Independent(dists.Normal(mean, std), 1),
-            RobustTanhTransform(),
-        )
-        sample = dist.rsample()
-        log_prob = dist.log_prob(sample).view(bsz, 1)
+            log_prob = dist.log_prob(sample).view(bsz, 1)
+
+            assert sample.shape == (bsz, self.action_dim)
+            assert log_prob.shape == (bsz, 1)
 
-        assert sample.shape == (bsz, self.action_dim)
-        assert log_prob.shape == (bsz, 1)
+            return sample, log_prob
 
-        return sample, log_prob
+        return torch.tanh(mean + std * epsilon)
 
 
 class GaussianMlpPolicy(nn.Module):
-    def __init__(self, observation_dim, action_dim, hidden_dims):
+    def __init__(self, observation_dim: int, action_dim: int, hidden_dims: list[int]):
```

Comment:
Python 3.8 does not support the `list` type syntax, has to be `List` until we drop 3.8.